eng/native/configurecompiler.cmake

```diff
@@ -195,6 +195,8 @@ elseif(CLR_CMAKE_HOST_SUNOS)
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fstack-protector")
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fstack-protector")
   add_definitions(-D__EXTENSIONS__)
+elseif(CLR_CMAKE_HOST_OSX)
+  add_definitions(-D_XOPEN_SOURCE)
```

Comment:
@janvorli, I was not about to reproduce the error you encountered locally. I am wondering if this definition is really necessary, as we already have it on coreclr level https://github.com/dotnet/runtime/blob/f9cf60113cf8734235ed42cd6a15218cdcc625d3/src/coreclr/clrdefinitions.cmake#L26
then we have it again on PAL level https://github.com/dotnet/runtime/blob/f9cf60113cf8734235ed42cd6a15218cdcc625d3/src/coreclr/pal/src/CMakeLists.txt#L244
the third one is on libunwind level (by our overwritten CMakeLists.txt file) https://github.com/dotnet/runtime/blob/f9cf60113cf8734235ed42cd6a15218cdcc625d3/src/coreclr/pal/src/libunwind/CMakeLists.txt#L74

Reply:
Maybe we can deduplicate those three in favor of this one.

Reply:
> I am wondering if this definition is really necessary, as we already have it on coreclr level

The definition in runtime/src/coreclr/pal/src/CMakeLists.txt only applies for DAC version of PAL. So it is not defined for regular PAL build. If we defined it in that file at line 11, we can remove it from the line 244 and from the runtime/src/coreclr/pal/src/libunwind/CMakeLists.txt. The definition in runtime/src/coreclr/clrdefinitions.cmake is useless. I've verified that the build works fine for me with these changes.

I am using SDK / clang from xcode 12 to build stuff, so I wonder if that's why it doesn't compile for me without these changes.

Reply:
Thanks. I have just realized that clrdefinitions are added after signal.cpp and friends are compiled.

Reply:
I have cleaned other occurrences and deleted an unused makefile with _XOPEN_SOURCE (this file is apparently unused for a while). The global definition is the only one in effect for non-mono parts.